### DISCUSSION TITLE
Default response_model on include_router

### DISCUSSION THREAD

**Discussion by zd4y** [fastapi/fastapi#9069] [Questions] [answered]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

How can I put a default `response_model` when including a router? For example, having this router:

```python
# routers/users.py

router = APIRouter()

@router.get('/{user_id}', response_model=User)
def get_user(user_id: int, db: Session = Depends(get_db)):
    return crud.get_user(db=db, user_id=user_id)


@router.put('/{user_id}', response_model=User)
def update_user(user_id: int, user: UserUpdate, db: Session = Depends(get_db)):
    return crud.update_user(db=db, user_id=user_id, user=user)
```

I am using the same `response_model` in all routes, so I might rather put it when including the router:

```python
# main.py

app.include_router(users.router, prefix='/users', default_response_model=User)
```

But this gives an error: `TypeError: include_router() got an unexpected keyword argument 'default_response_model'` for both `default_response_model` and `response_model`.

Is there any reason why this is not supported?

**Comment by dbanty** [comment]
I believe you can put the response_model in the declaration of `APIRouter()` instead of where you register it.

```python
router = APIRouter(response_model=User)
```

Not at a computer to test that on, but hopefully it does the trick 😅

**Comment by zd4y** [comment]
Same error: `TypeError: __init__() got an unexpected keyword argument 'response_model'`

**Comment by dbanty** [comment]
Yeah that's my bad, I really thought this was already a feature 😨.  Seems like a good feature to implement for sure.

**Comment by mpaccione** [comment]
I have the same problem! All of my endpoints return JSON (except for a string for some dev only apis). Used the router to have logical seperation of concerns but it seems like I can't find very good documentation on this issue?

**Comment by tobiasbrodersen** [comment]
Hey, hit the same snag today.

Has this been implemented? Could this just be a lingering discussion? 🙏

**Comment by YuriiMotov** [comment, marked as answer]
You can specify default model using `responses` parameter of `APIRouter` or even entire `FastAPI` app:

```
from fastapi import APIRouter, FastAPI
from pydantic import BaseModel


class DefaultResponseSchema(BaseModel):
    status: str
    msg: str

class UserSchema(BaseModel):
    name: str


router = APIRouter(prefix="/router")

# Route with default response model for status code 200
@router.get("/")
def get_index():
    return {"status": "Ok", "msg": "there is a message"}


# Route with custom response model for status code 200
@router.get(
    "/user",
    responses={
        200: {"model": UserSchema},
    },
)
def get_user():
    return {"name": "Yurii"}


# Set default response model for status code 200 for entire app
app = FastAPI(
    responses={
        200: {"model": DefaultResponseSchema},
    },
)

app.include_router(router=router)
```

The drawback is that to redefine response model for particular route you have to also use `responses`..
